iwyu_ast_util.cc

```diff
@@ -250,6 +253,38 @@ static const RedeclarableTemplateDecl* GetRedeclNotSpecifyingAnyDefArg(
   return nullptr;
 }
 
+template <typename T>
+static void PrintExplicitSpecTplArgs(const T* spec,
+                                     const PrintingPolicy& printing_policy,
+                                     raw_ostream& ostream) {
+  if (spec->isExplicitSpecialization()) {
+    ostream << '<';
+    const TemplateArgumentList& args = spec->getTemplateArgs();
+    const TemplateParameterList* params =
+        spec->getSpecializedTemplate()->getTemplateParameters();
+    for (unsigned i = 0, ie = args.size(); i < ie; ++i) {
+      if (i > 0)
+        ostream << ", ";
+      bool include_type = TemplateParameterList::shouldIncludeTypeForArgument(
+          printing_policy, params, i);
+      const TemplateArgument& arg = args.get(i);
+      // Clang embraces arguments corresponding to a parameter pack in angle
+      // brackets, so cannot use the default 'print' for them.
+      if (arg.getKind() == TemplateArgument::Pack) {
+        ArrayRef<TemplateArgument> pack_args = arg.getPackAsArray();
+        for (size_t j = 0, je = pack_args.size(); j < je; ++j) {
+          if (j > 0)
```

Comment:
And using the ListSeparator here would make it possible to simplify the loop into a range loop, I think.